File: Linguaggio_Alto_Livello/JavaScript/Hard/prime_sieve.js
const limit = 50;
const isPrime = Array(limit + 1).fill(true);
isPrime[0] = false;
isPrime[1] = false;

for (let number = 2; number * number <= limit; number += 1) {
  if (isPrime[number]) {
    for (let multiple = number * number; multiple <= limit; multiple += number) {
      isPrime[multiple] = false;
    }
  }
}

const primes = isPrime
  .map((prime, index) => (prime ? index : null))
  .filter((value) => value !== null);

console.log("Numeri primi fino a 50:");
console.log(primes);

File: Linguaggio_Alto_Livello/Python/Hard/prime_sieve.py
limit = 50
is_prime = [True] * (limit + 1)
is_prime[0] = is_prime[1] = False

for number in range(2, int(limit ** 0.5) + 1):
    if is_prime[number]:
        for multiple in range(number * number, limit + 1, number):
            is_prime[multiple] = False

primes = [n for n, prime in enumerate(is_prime) if prime]
print("Numeri primi fino a 50:")
print(primes)
